mzLib/FlashLFQ/FlashLFQResults.cs

```diff
@@ -159,16 +163,34 @@ public void CalculatePeptideResults()
                         double alreadyRecordedIntensity = PeptideModifiedSequences[sequence].GetIntensity(filePeaks.Key);
                         double fractionAmbiguous = ambiguousPeak.Intensity / (alreadyRecordedIntensity + ambiguousPeak.Intensity);
 
-                        if (fractionAmbiguous > 0.3)
+                        if (quantifyAmbiguousPeptides)
+                        {
+                            // If the peptide intensity hasn't been recorded, the intensity is set equal to the intensity of the ambiguous peak
+                            if (Math.Abs(alreadyRecordedIntensity) < 0.01)
+                            {
+                                PeptideModifiedSequences[sequence].SetDetectionType(filePeaks.Key, DetectionType.MSMSAmbiguousPeakfinding);
+                                PeptideModifiedSequences[sequence].SetIntensity(filePeaks.Key, ambiguousPeak.Intensity);
+                            }
+                            // If the peptide intensity has already been recorded, that value is retained. 
+                            else if (fractionAmbiguous > 0.3)
+                            {
+                                PeptideModifiedSequences[sequence].SetDetectionType(filePeaks.Key, DetectionType.MSMSAmbiguousPeakfinding);
+                            }
```

Comment:
Suppose you have two peaks. Peak1 is identified as PEPTIDEK, and Peak2 is ambiguously identified as PEPTIDEK | PPETIDEK. Then, fractionAmbiguous is equal to Peak2.Intensity / (Peak1.Intensity + Peak2.Intensity). If Peak2 is > 2.33 times more intense than Peak1, then fractionAmbiguous is greater than 0.3.

Previously when this occurred, PEPTIDEK's detection type was changed to MSMSAmbiguousPeakFinding, and its intensity was changed from Peak1.Intensity to 0.  

I can't find any tests that cover the scenario where a peptide had an intensity value (i.e., was found in Peak1), was associated with a different ambiguous peak, and then had the intensity value changed. Implementing this would be onerous, because it would require a new slice.mzml to be generated